src/compiler/transformers/es2015.ts

```diff
@@ -1782,7 +1786,7 @@ namespace ts {
             const func = factory.createFunctionExpression(
                 /*modifiers*/ undefined,
                 /*asteriskToken*/ undefined,
-                /*name*/ undefined,
+                /*name*/ name,

```

Comment:
I think this needs a check based on if the function body uses this identifier or not. Is there a utility function I can call to get the list of all identifiers used by the function body?